### PR TITLE
Af 543 Aerodrome Finance AM

### PR DESCRIPTION
**Pull request by acollette**

### CODE REVIEW

File: src/asset-modules/Aerodrome-Finance/interfaces/IAeroRouter.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: MIT
+ */
+pragma solidity 0.8.22;
+
+interface IAeroRouter {
```
Comment (Thomas-Smets) [inline src/asset-modules/Aerodrome-Finance/interfaces/IAeroRouter.sol:9]: Is this necessary for the asset modules or only for tests?
Move interfaces only for tests to other folders

File: src/asset-modules/Aerodrome-Finance/AerodromeVolatileAM.sol [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    /*///////////////////////////////////////////////////////////////
+                        ASSET INFORMATION
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Checks for a token address and the corresponding id if it is allowed.
+     * @param asset The contract address of the asset.
+     * @return allowed A boolean, indicating if the asset is allowed.
+     */
+    function isAllowed(address asset, uint256) public view override returns (bool allowed) {
```
Comment (Thomas-Smets) [inline src/asset-modules/Aerodrome-Finance/AerodromeVolatileAM.sol:94]: Think we should also check for non-deposited assets if the underlying assetsmatch, as in uniV3

File: src/asset-modules/Aerodrome-Finance/StakedAerodromeAM.sol [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    /*///////////////////////////////////////////////////////////////
+                            ASSET MANAGEMENT
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Adds a new Staked Aerodrome Finance pool to the StakedAerodromeAM.
+     * @param pool The contract address of the Aerodrome Finance pool.
+     * @param gauge The contract address of the gauge to stake the Aerodrome Finance LP.
+     */
+    function addAsset(address pool, address gauge) external {
```
Comment (Thomas-Smets) [inline src/asset-modules/Aerodrome-Finance/StakedAerodromeAM.sol:57]: How do we know that gauge is not malicious?

Comment (acollette) [reply src/asset-modules/Aerodrome-Finance/StakedAerodromeAM.sol:68]: added the isGauge check on Voter.sol from Aerodrome

File: test/fuzz/asset-modules/SlipstreamAM/GetPosition.fuzz.t.sol [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        // Given: Position is valid.
+        position = givenValidPosition(position);
+
+        // And: Liquidity is non-zero.
+        position.liquidity = uint128(bound(position.liquidity, 1, type(uint128).max));
+
+        // And: State is persisted.
+        nonfungiblePositionManagerMock.setPosition(address(poolStable1Stable2), assetId, position);
+
+        // And: The Uniswap V3 position is added to the Asset Module.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/GetPosition.fuzz.t.sol:54]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            (token0, token1) = (token1, token0);
+            (vars.decimals0, vars.decimals1) = (vars.decimals1, vars.decimals0);
+            (vars.priceToken0, vars.priceToken1) = (vars.priceToken1, vars.priceToken0);
+        }
+
+        // Avoid divide by 0 in next line.
+        vm.assume(vars.priceToken1 > 0);
+        // Cast to uint160 will overflow, not realistic.
+        vm.assume(vars.priceToken0 / vars.priceToken1 < 2 ** 128);
+        // Check that sqrtPriceX96 is within allowed Uniswap V3 ranges.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol:61]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        uint160 sqrtPriceX96 = slipstreamAM.getSqrtPriceX96(
+            vars.priceToken0 * 10 ** (18 - vars.decimals0), vars.priceToken1 * 10 ** (18 - vars.decimals1)
+        );
+
+        vm.assume(sqrtPriceX96 >= 4_295_128_739);
+        vm.assume(sqrtPriceX96 <= 1_461_446_703_485_210_103_287_273_052_203_988_822_378_723_970_342);
+
+        // Create Uniswap V3 pool initiated at tickCurrent with cardinality 300.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol:69]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        // This is because there might be some small differences due to rounding errors.
+        (,,,,,,, uint128 liquidity_,,,,) = nonfungiblePositionManager.positions(tokenId);
+        (uint256 amount0, uint256 amount1) = LiquidityAmounts.getAmountsForLiquidity(
+            sqrtPriceX96,
+            TickMath.getSqrtRatioAtTick(vars.tickLower),
+            TickMath.getSqrtRatioAtTick(vars.tickUpper),
+            liquidity_
+        );
+
+        // Overflows Uniswap libraries, not realistic.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol:89]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        registryExtension.setRiskParametersOfDerivedAM(
+            address(creditorUsd), address(slipstreamAM), type(uint112).max, uint16(riskFactorUniV3)
+        );
+        vm.stopPrank();
+
+        // First take minimum of each risk factor.
+        uint256 expectedCollFactor = collFactor0 < collFactor1 ? collFactor0 : collFactor1;
+        uint256 expectedLiqFactor = liqFactor0 < liqFactor1 ? liqFactor0 : liqFactor1;
+
+        // Next apply risk factor for uniswap V3.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/GetValue.fuzz.t.sol:157]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+        vm.assume(isWithinAllowedRange(tickLower));
+        vm.assume(isWithinAllowedRange(tickUpper));
+
+        vm.assume(liquidity > 0);
+
+        // Calculate and check that tick current is within allowed ranges.
+        uint160 sqrtPriceX96 = uint160(calculateAndValidateRangeTickCurrent(priceToken0, priceToken1));
+        vm.assume(isWithinAllowedRange(TickMath.getTickAtSqrtRatio(sqrtPriceX96)));
+
+        // Create Uniswap V3 pool initiated at tickCurrent with cardinality 300.
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol:70]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol [hunk 10]
```diff
@@ -0,0 +1,460 @@
...
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol:109]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol [hunk 11]
```diff
@@ -0,0 +1,460 @@
...
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol:147]: Uni -> slipstream

File: test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol [hunk 12]
```diff
@@ -0,0 +1,460 @@
...
```
Comment (j-vp) [inline test/fuzz/asset-modules/SlipstreamAM/ProcessIndirectDeposit.fuzz.t.sol:257]: Uni -> slipstream

**Pull request merged**